backend/demo_main.py

```
@@ -0,0 +1,261 @@
+            raise HTTPException(status_code=400, detail="Only PDF and TXT files are supported")
+        
+        # Generate unique ID
+        document_id = str(uuid.uuid4())
+        
+        # Save file temporarily for demo
+        temp_path = f"/tmp/{document_id}_{file.filename}"
+        try:
+            with open(temp_path, "wb") as buffer:
+                content = await file.read()
+                buffer.write(content)
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)